doc/news/changes/minor/20170809ArndtMaierPelteret

```diff
@@ -0,0 +1,4 @@
+Fixed: Debugged and fixed a long standing linker error due to missing
+instantiations of ShapeInfo::reinit variants.
```

Comment:
@bangerth It is fairly generic - only depends on whether the optimizer inlines reinit<> in another compilation unit (we reproduced on mac and linux with clang, and two gcc versions).